src/test/kotlin/no/skatteetaten/aurora/mokey/service/ApplicationDataServiceOpenShiftTest.kt

```diff
@@ -0,0 +1,73 @@
+package no.skatteetaten.aurora.mokey.service
+
+import assertk.assert
+import assertk.assertions.contains
+import assertk.assertions.hasSize
+import assertk.assertions.isEqualTo
+import io.mockk.clearMocks
+import io.mockk.every
+import io.mockk.mockk
+import no.skatteetaten.aurora.mokey.DeploymentConfigDataBuilder
+import no.skatteetaten.aurora.mokey.ImageDetailsDataBuilder
+import no.skatteetaten.aurora.mokey.PodDetailsDataBuilder
+import no.skatteetaten.aurora.mokey.ProjectDataBuilder
+import no.skatteetaten.aurora.mokey.ReplicationControllerDataBuilder
+import no.skatteetaten.aurora.mokey.model.AuroraStatus
+import no.skatteetaten.aurora.mokey.model.AuroraStatusLevel
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class ApplicationDataServiceOpenShiftTest {
+
+    private val openShiftService = mockk<OpenShiftService>()
+    private val auroraStatusCalculator = mockk<AuroraStatusCalculator>()
+    private val podService = mockk<PodService>()
+    private val imageService = mockk<ImageService>()
+    private val applicationDataServiceOpenShift = ApplicationDataServiceOpenShift(
+            openShiftService,
+            auroraStatusCalculator,
+            podService,
+            imageService
+    )
+
+    @BeforeEach
+    fun setUp() {
+        clearMocks(openShiftService, auroraStatusCalculator, podService, imageService)
+    }
+
+    @Test
+    fun `find all affiliations`() {
+        val project = ProjectDataBuilder().build()
+        every { openShiftService.projects() } returns listOf(project)
+        val affiliations = applicationDataServiceOpenShift.findAllAffiliations()
+
+        assert(affiliations) {
+            hasSize(1)
+            contains("affiliation")
+        }
+    }
+
+    @Test
+    fun `find application data by id`() {
+        val dc = DeploymentConfigDataBuilder().build()
+        every { openShiftService.deploymentConfigs("affiliation") } returns listOf(dc)
+        val replicationController = ReplicationControllerDataBuilder().build()
+        every { openShiftService.rc("namespace", "app-name-1") } returns replicationController
```

Comment:
Jeg ser du konsekvent bruker konstanter her istedenfor å refererere til disse verdiene i builderne. Jeg antar at du har en grunn til dette, men jeg klarte ikke helt tenke ut hva fordelen evt. er. Blir det ikke både mer robuste tester og mindre redundans dersom man matcher mot verdiene i builderne her?

Reply:
Godt spørsmål!

Jeg er litt usikker her rett og slett på hva jeg syns er best.
Vi kan gjøre mock'ene mer generiske, så slipper man dette problemet. Litt av grunnen til at jeg pleier å ikke gjøre det er at det kan snike seg inn feil fordi mockene bare godtar alt.
Det andre er at vi kunne hentet ut fra builder eller fra selve dto'en. Fra dto'en i dette tilfellet blir litt tungvindt fordi det ligger så langt inn i strukturen. Men kanskje man kunne hentet det fra builderen som du foreslår. Jeg hadde også laget en variant med companion object, for å få konstanter på verdiene. Men kanskje du vet om en bedre variant?

Reply:
jeg så du har fjernet private på en del av "val" ene i builderne.
Kanskje vi kan gjøre noe sånt?
```
val dcBuilder = DeploymentConfigDataBuilder()
every { openShiftService.deploymentConfigs(dcBuilder.dcAffiliation) } returns listOf(dcBuilder.build())
```

Reply:
Ja, jeg fjernet val, men det var utilsiktet å committe (:man_facepalming: ) Jeg eksperimenterte litt med å hente verdiene ut fra builderne i testene. Jeg endte opp med å like builderne best siden jeg opplever det som i litt større grad parametre til testen enn selve dto'en. Da tenker jeg også at det må være greit å matche mot dem i assertene?

Reply:
Faktisk så har konstanter i builderne en ulempe her kontra å matche mot builder-instansenes properties dersom man kopierer en builder med alternative verdier. Jeg syntes det er helt greit med public properties på builderne her jeg, altså.

Reply:
jeg er enig, syns beste er å ha public properties på builder instansen.
Jeg oppdaterer testene.